docker-compose.yml

```diff
@@ -1,15 +1,50 @@
 version: '3.7'
+
 services:
-  server:
-    container_name: shoe_project_server
+  build-setup:
+    image: shoe_project_image
     build:
       context: .
-      dockerfile: $PWD/docker/Dockerfile.dev
+      dockerfile: './docker/Dockerfile.dev'
+    environment:
+      DATABASE_HOST: postgres
+    command: './shoe-project seed'
+    depends_on:
+      - postgres
+
+  server:
+    image: shoe_project_image
+    container_name: shoe_project_server
+    environment:
+      DATABASE_HOST: postgres
     ports:
+      - '42967:42967' # Hot Module Reloading for frontend
       - '8900:8900'
+    command:
+      - /bin/sh
+      - -c
+      - |
+          cd ui && yarn watch &
+          CompileDaemon --command="./shoe-project server"
+    volumes:
+      - './:/app'
+    depends_on:
+      - build-setup
+      - postgres
+
+  frontend:
+    image: shoe_project_image
+    container_name: shoe_project_frontend
+    ports:
       - '42967:42967' # Hot Module Reloading for frontend
-    environment:
-      DATABASE_HOST: postgres
+      - '1234:1234'
+    working_dir: '/app/ui'
+    command:
+      - /bin/sh
+      - -c
+      - |
+          yarn watch &
+          yarn start
```

Comment:
Why run both `yarn start` and `yarn watch`?

Isn't this duplicating how we serve assets? Don't really see a need for node running it's own server.

If we decide to move this way though, it might make sense for us to server side render with a small express server. That way we can hydrate and make first page load way faster (we don't have to do it rn, we can do it whenever, even next term). See: https://reactjs.org/docs/react-dom-server.html

Reply:
+1 on moving it to another service tho

Reply:
I made `frontend` run its own node server because this service can run independently and frontend developers don't need go server to run. When we are running all services, we can just ignore that node server